src/matchers/Emoji.js

```diff
@@ -90,20 +99,22 @@ export default class EmojiMatcher extends Matcher<EmojiOptions> {
    */
   onAfterParse(content: ParsedNodes): ParsedNodes {
     // When a single `Emoji` is the only content, enlarge it!
-    if (content.length === 1) {
-      let item = content[0];
-
-      if (typeof item !== 'string' && React.isValidElement(item) && item.type === Emoji) {
-        item = (item: React.Element<*>);
-
-        return [
-          React.cloneElement(item, {
-            ...item.props,
-            enlargeEmoji: true,
-          }),
-        ];
-      }
+    if (content.length > this.options.enlargeUpTo
+        || !content.every(item => typeof item !== 'string'
```

Comment:
Let's create a method for the callback instead of constantly recreating it.

```js
isAnEmoji(item) {
  return (typeof item !== 'string' && React.isValidElement(item) && item.type === Emoji)
}
```

```
!content.every(this.isAnEmoji)
```